OData/src/System.Web.OData/OData/Query/Expressions/UriFunctionsToClrBinder.cs

```diff
@@ -0,0 +1,198 @@
+
+            lock (Locker)
+            {
+                return MethodLiteralSignaturesToMethodInfo.Remove(methodLiteralSignature);
+            }
+        }
+
+        #endregion
+
+        #region Internal Methods
+
+        internal static bool TryGetMethodInfo(string functionName, IEnumerable<Type> methodArgumentsType, out MethodInfo methodInfo)
+        {
+            // Validation
+            if (string.IsNullOrEmpty(functionName))
+            {
+                throw Error.ArgumentNull("functionName");
+            }
+
+            if (methodArgumentsType == null)
+            {
+                throw Error.ArgumentNull("methodArgumentsType");
+            }
+
+            // Get literal description of the method
+            string methodLiteralSignature = GetMethodLiteralSignature(functionName, methodArgumentsType);
+
+            lock (Locker)
+            {
+                return MethodLiteralSignaturesToMethodInfo.TryGetValue(methodLiteralSignature, out methodInfo);
+            }
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        /// <summary>
+        /// Get a string describing the given method.
+        /// </summary>
+        /// <param name="methodName"></param>
+        /// <param name="methodInfo"></param>
+        /// <returns></returns>
+        private static string GetMethodLiteralSignature(string methodName, MethodInfo methodInfo)
+        {
+            // Get the arguments type of the given MethodInfo
+            IEnumerable<Type> methodArgumentsType = methodInfo.GetParameters().Select(parameter => parameter.ParameterType);
+
+            // If method is not static - instance, the declaring type is the first argument. 
+            if (!methodInfo.IsStatic)
+            {
+                methodArgumentsType = new Type[] { methodInfo.DeclaringType }.Concat(methodArgumentsType);
+            }
+
+            return GetMethodLiteralSignature(methodName, methodArgumentsType);
+        }
+
+        /// <summary>
+        /// Creates a string describing the function signature.
+        /// 'methodName(argTypeName1,argTypeName2,argTypeName3..)'
+        /// </summary>
+        /// <param name="methodName"></param>
+        /// <param name="methodArgumentsType"></param>
+        /// <returns></returns>
+        private static string GetMethodLiteralSignature(string methodName, IEnumerable<Type> methodArgumentsType)
+        {
+            StringBuilder builder = new StringBuilder();
+            string descriptionSeparator = "";
+            builder.Append(descriptionSeparator);
+            descriptionSeparator = "; ";
+
+            string parameterSeparator = "";
+            builder.Append(methodName);
+            builder.Append('(');
+            foreach (Type type in methodArgumentsType)
+            {
+                builder.Append(parameterSeparator);
+                parameterSeparator = ", ";
```

Comment:
In this line I change the parameterSeperator value from "" to ", ".  Because after the first loop we need to assign a comma between each type. Another way to do this is to hold an index variable and assin comma  if the index is above 0 otherwise assign nothing.
Another way is to do the first loop outside the for loop, and do the rest in the for loop.
Choose your ideal solution.